src/AppBundle/Controller/ArticleController.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity\Article;
+use AppBundle\Form\ArticleType;
+use AppBundle\Security\ArticleVoter;
+use Symfony\Component\Form\Form;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Annotation\Route;
+
+class ArticleController extends BaseController
+{
+    /**
+     * @param $request
+     * @Route("/article/new", name="new_article")
+     *
+     * @return Response
+     */
+    public function newAction(Request $request)
+    {
+        $form = $this->createForm(ArticleType::class);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $article = $form->getData();
+
+            $this->denyAccessUnlessGranted(ArticleVoter::CREATE_ARTICLE, $article);
+            if (!$article->getImage()) {
+                $article->setImage('50708_1280x720-318x180.jpg');
+            }
+
+            $article->setAuthor($this->getUser()
+                ->getAuthor()
+            );
+
+            $this->em()->persist($article);
+
+            $this->em()->flush();
+
+            $this->get('app.notifier')
+                ->newArticleNotify($article, $article->getAuthor());
+
+            return $this->redirectToRoute('homepage');
+        }
+
+        return $this->render(':Article:new.html.twig', [
+            'articleType' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{page}", name="homepage", requirements={"page": "\d+"})
+     *
+     * @param $request
+     * @param $page
+     *
+     * @return Response
+     */
+    public function listAction(Request $request, $page = 1)
+    {
+        $articles = $this->em()->getRepository('AppBundle:Article')
+            ->findAllPublished();
+
+        if (!$articles) {
+            throw new NotFoundHttpException('Noooo!');
+        }
+
+        $pagination = $this->pagination($articles,
+            $request->query->getInt('page', $page), 5);
+
+        return $this->render('Article/list.html.twig', [
+        'articles' => $pagination,
+    ]);
+    }
+
+    /**
+     * @param $request
+     * @param $article
+     * @Route("/article/edit/{id}", name="edit_article")
+     *
+     * @return Response
+     */
+    public function editAction(Request $request, Article $article)
+    {
+        $this->denyAccessUnlessGranted(ArticleVoter::EDIT_ARTICLE, $article);
+
+        $form = $this->createForm(ArticleType::class, $article);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $article = $form->getData();
```

Comment:
You may not create this new variable but use old variable $article, it's already modified